tests/jerry/regression-test-issue-3779.js

```diff
@@ -0,0 +1,23 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var a = [1, 2, 3];
+var idx = a.lastIndexOf(50, {
+    valueOf: function() {
+        // Trigger removing elements from fast array.
+        a.length = 0;
+    }
+})
```

Comment:
Please add the following testcase as well to ensure that the added element during the callback does not affect the original limit of the lookup.

```js
var a = [1, 2, 3, 4];
var idx = a.lastIndexOf(4, {
    valueOf: function() {
        a.push(4)
    }
})

assert (idx === -1);
```

Reply:
True. Thanks, I've updated the pr.